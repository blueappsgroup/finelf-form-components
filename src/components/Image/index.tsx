import React from 'react'
import styled from 'styled-components'
import { device } from '../../consts/sizes'
import { ImageProps } from '../../types'

type StyledProps = {
  hideOnMobile?: boolean
  disableShadow?: boolean
  theme: {
    imageContainerPadding: string
    imageBoxShadow: string
    imageContainerTextTopPosition: string
    imageContainerTextFontSize: string
    imageContainerTextFontWeight: string
  }
}

// prettier-ignore
const StyledContainer = styled.div<StyledProps>`
  position: relative;
  display: ${(props: StyledProps): string => props.hideOnMobile ? 'none' : 'flex'};
  justify-content: center;
  width: 100%;
  padding: ${(props): string => props.theme.imageContainerPadding};
  margin: 0 auto;
  align-items: flex-start;
  box-sizing: border-box;
  ${(props: StyledProps) =>
    !props.disableShadow &&
    'padding-right: 30px;'}
  
  img {
    max-width: 100%;  
    box-shadow: ${(props: StyledProps): string => props.disableShadow ? 'none' : props.theme.imageBoxShadow};
  }
  
  span {
    position: absolute;
    top: ${(props): string => props.theme.imageContainerTextTopPosition};
    font-size: ${(props): string => props.theme.imageContainerTextFontSize};
    font-weight: ${(props): string => props.theme.imageContainerTextFontWeight};
  }
  
  @media ${device.laptopL} { 
    display: flex;
  }
`

const Image: React.FC<ImageProps> = ({
  src,
  alt,
  text,
  hideOnMobile,
  disableShadow,
}) => (
  <StyledContainer hideOnMobile={hideOnMobile} disableShadow={disableShadow}>
    <img src={src} alt={alt} />
    {text && <span>{text}</span>}
  </StyledContainer>
)

export default Image
