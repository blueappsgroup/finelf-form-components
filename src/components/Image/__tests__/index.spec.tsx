import { render } from '@testing-library/react'
import * as React from 'react'
import Image from '../'

describe('<Image />', () => {
  it('image matches snapshot', () => {
    const documentBody = render(<Image src="" alt="image" hideOnMobile />)
    const { baseElement } = documentBody

    expect(baseElement).toMatchSnapshot()
  })
})
