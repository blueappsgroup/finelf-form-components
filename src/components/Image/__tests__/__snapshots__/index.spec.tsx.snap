// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> image matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bdnylx jbLzLu"
    >
      <img
        alt="image"
        src=""
      />
    </div>
  </div>
</body>
`;
